networks/rpc/handler.go

```diff
@@ -427,6 +428,23 @@ func (h *handler) handleSubscribe(cp *callProc, msg *jsonrpcMessage) *jsonrpcMes
 func (h *handler) runMethod(ctx context.Context, msg *jsonrpcMessage, callb *callback, args []reflect.Value) *jsonrpcMessage {
 	result, err := callb.call(ctx, msg.Method, args)
 	if err != nil {
+		if _, ok := err.(*statedb.MissingNodeError); ok {
```

Comment:
```go
if UpstreamArchiveEN != "" && shouldRequestUpstream(result, err) {
    return requestUpstream(ctx, msg, args)
}
```

```go
func shouldRequestUpstream(result, err) bool {
    switch err.(type) {
    case *statedb.MissingNodeError:
        return true
    default:
        return false
    }
}

func requestUpstream(ctx, msg, args) *jsonrpcMessage {
}
```

Reply:
@blukat29 
`UpstreamArchiveEN != ""` is checked at `url.Parse(rawurl)` of `DialContext` method.
https://github.com/sjnam/klaytn/blob/reliable-archive-en/networks/rpc/client.go#L176-L180
But, it's good idea to explicitly check for `UpstreamArchiveEN != ""`.

Reply:
@sjnam I see. But if we dial when `UpstreamArchiveEN == ""`, then (depending on the implementation) the outer RPC returns "URL parse error", not "missing node error". So I think the explicit check is necessary.